### PR TITLE
FIX: move post events into separate tables

### PR DESCRIPTION
**Pull request by lis2**
Instead of relying on scheduled jobs, a new table is introduced.

Event `starts_at` and `ends_at` are renmaned to `original_starts_at` and `original_ends_at` and those are configuration values extracted from Post.

`EventDates` are actual event dates and `MonitorEventDates` job is responsible for creating reminders, triggering events and creating new recurrent date.

### CODE REVIEW

**Comment by lis2** [comment]
@jjaffeux I am still cleaning and checking if everything works as expected. However, I wanted to create that draft to show you a concept and get your opinion about that approach.

**Review by jjaffeux — APPROVED** [review]

File: spec/jobs/scheduled/monitor_event_dates_spec.rb [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+describe DiscourseCalendar::MonitorEventDates do
+  fab!(:post_1) { Fabricate(:post) }
+  fab!(:post_2) { Fabricate(:post) }
+  fab!(:past_event) { Fabricate(:event, post: post_1, original_starts_at: 7.days.after, original_ends_at: 7.days.after + 1.hour, reminders: '15.minutes,1.hours') }
+  let(:past_date) { past_event.event_dates.first }
+  fab!(:future_event) { Fabricate(:event, post: post_2, original_starts_at: 14.days.after, original_ends_at: 14.days.after + 1.hour) }
+  let(:future_date) { future_event.event_dates.first }
+
+  context '#send_reminder' do
+    # Jobs.enqueue(:discourse_post_event_send_reminder, event_id: event_date.event.id, reminder: reminder)
```
Comment (jjaffeux) [inline spec/jobs/scheduled/monitor_event_dates_spec.rb:16]: do we want to keep tat comment ?

File: spec/jobs/scheduled/destroy_past_events_spec.rb [hunk 2]
```diff
@@ -48,6 +48,15 @@
     expect(Post.find_by(id: post.id)).not_to eq(nil)
   end
 
+  it "will destroy expired standalone events" do
+    topic = Fabricate(:topic)
+    event = CalendarEvent.create!(topic: calendar_post.topic, start_date: 10.years.ago)
```
Comment (jjaffeux) [inline spec/jobs/scheduled/destroy_past_events_spec.rb:53]: I'm surprise we have CalendarEvent here, this is a temporary artifact used after the migration from custom fields to tables but is not actually related to events, but only to topic calendars (let me know if that doesn't make sense).

**Comment by k4rtik** [comment]
Hi

It looks like since this PR merge, there is a bug introduced. When I create an event it shows a preview on the right, but the event UI does not show on saving the post.

Here's what I see in logs:

```
Message (31 copies reported)

Job exception: unknown attribute 'original_starts_at' for DiscoursePostEvent::Event.


Backtrace

/var/www/discourse/vendor/bundle/ruby/2.6.0/gems/activemodel-6.0.3.3/lib/active_model/attribute_assignment.rb:52:in `_assign_attribute'
/var/www/discourse/vendor/bundle/ruby/2.6.0/gems/activemodel-6.0.3.3/lib/active_model/attribute_assignment.rb:43:in `block in _assign_attributes'
/var/www/discourse/vendor/bundle/ruby/2.6.0/gems/activemodel-6.0.3.3/lib/active_model/attribute_assignment.rb:42:in `each'
/var/www/discourse/vendor/bundle/ruby/2.6.0/gems/activemodel-6.0.3.3/lib/active_model/attribute_assignment.rb:42:in `_assign_attributes'
/var/www/discourse/vendor/bundle/ruby/2.6.0/gems/activerecord-6.0.3.3/lib/active_record/attribute_assignment.rb:21:in `_assign_attributes'
/var/www/discourse/vendor/bundle/ruby/2.6.0/gems/activemodel-6.0.3.3/lib/active_model/attribute_assignment.rb:35:in `assign_attributes'
/var/www/discourse/vendor/bundle/ruby/2.6.0/gems/activerecord-6.0.3.3/lib/active_record/persistence.rb:634:in `block in update!'
/var/www/discourse/vendor/bundle/ruby/2.6.0/gems/activerecord-6.0.3.3/lib/active_record/transactions.rb:375:in `block in with_transaction_returning_status'
/var/www/discourse/vendor/bundle/ruby/2.6.0/gems/activerecord-6.0.3.3/lib/active_record/connection_adapters/abstract/database_statements.rb:280:in `block in transaction'
/var/www/discourse/vendor/bundle/ruby/2.6.0/gems/activerecord-6.0.3.3/lib/active_record/connection_adapters/abstract/transaction.rb:280:in `block in within_new_transaction'
```

**Comment by k4rtik** [comment]
Update: it seems to have worked when I did a full upgrade using ssh.

**Pull request merged**